File: app/src/main/java/wj/com/myplayer/View/Activity/MainMusic/LocalMusicActivity.java
package wj.com.myplayer.View.Activity.MainMusic;

import android.content.ComponentName;
import android.content.Intent;
import android.content.ServiceConnection;
import android.os.Bundle;
import android.os.IBinder;
import android.support.v4.app.FragmentManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;

import java.util.List;

import wj.com.myplayer.Bean.MusicBean;
import wj.com.myplayer.Config.BaseMultipleActivity;
import wj.com.myplayer.Constant.SPConstant;
import wj.com.myplayer.DaoDB.MediaDaoManager;
import wj.com.myplayer.DaoDB.MediaEntity;
import wj.com.myplayer.R;
import wj.com.myplayer.Utils.SPUtils;
import wj.com.myplayer.View.Fragment.PlayFragment;
import wj.com.myplayer.View.adapter.MusicListAdapter;

public class LocalMusicActivity extends BaseMultipleActivity implements BaseQuickAdapter.OnItemChildClickListener, BaseQuickAdapter.OnItemClickListener {

    private RecyclerView mLocalMusicRv;
    private MusicListAdapter adapter;
    private List<MediaEntity> datalist;
    private MusicService.MusicBinder mBinder;

    private PlayFragment playFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_local_music);
        initView();
    }

    @Override
    public void initTitle() {
        mTitleCenterTv.setText("本地音乐");
    }

    public void initView() {
        mLocalMusicRv = (RecyclerView) findViewById(R.id.local_music_rv);
        datalist = MediaDaoManager.getInstance().getAllList();
        adapter = new MusicListAdapter(datalist);
        mLocalMusicRv.setLayoutManager(new LinearLayoutManager(this));
        mLocalMusicRv.setAdapter(adapter);
        adapter.setOnItemChildClickListener(this);
        adapter.setOnItemClickListener(this);

        FragmentManager fragmentManager = getSupportFragmentManager();
        playFragment = (PlayFragment) fragmentManager.findFragmentById(R.id.music_play_lay);

        Intent startMusicIntent = new Intent(this,MusicService.class);
        bindService(startMusicIntent,connection,BIND_AUTO_CREATE) ;
        startService(startMusicIntent);
    }

    private ServiceConnection connection = new ServiceConnection() {
        @Override
        public void onServiceConnected(ComponentName name, IBinder service) {
            mBinder = (MusicService.MusicBinder) service;
            playFragment.setBinder(mBinder);
        //    mBinder.play();
        }

        @Override
        public void onServiceDisconnected(ComponentName name) {

        }
    };

    @Override
    public boolean onItemChildClick(BaseQuickAdapter adapter, View view, int position) {
        /*if (view.getId() == R.id.item_music_more){
            String path = datalist.get(position).path;
            mBinder.play(path);
            return true;
        }*/
        return false;
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unbindService(connection);
    }

    @Override
    public void onItemClick(BaseQuickAdapter adapter, View view, int position) {
        if (mBinder != null){
            MediaEntity entity = datalist.get(position);
            mBinder.play(entity);
            MusicBean bean = new MusicBean();
            bean.setId(entity.getId());
            bean.setPath(entity.getPath());
            bean.setAlbum_id(entity.getAlbum_id());
            bean.setTitle(entity.getTitle());
            bean.setDisplay_name(entity.getDisplay_name());
            bean.setCover(entity.cover);
            bean.setArtist(entity.artist);
            bean.setSize(entity.size);
            bean.setSinger(entity.singer);
            bean.setDuration(entity.duration);
            bean.setAlbums(entity.albums);
            SPUtils.putObject(this, SPConstant.LAST_PALY_MUSIC,bean);
        }
    }
}

File: module_local/src/moduledebug/java/com/hjl/module_local/LocalMusicActivity.kt
package com.hjl.module_local

import android.Manifest
import android.content.ComponentName
import android.content.Intent
import android.content.ServiceConnection
import android.os.Bundle
import android.os.IBinder
import android.support.v4.app.Fragment
import android.view.View
import com.hjl.commonlib.adapter.LazyFragmentPagerAdapter
import com.hjl.commonlib.base.BaseMultipleActivity
import com.hjl.commonlib.utils.PermissionsUtiles
import com.hjl.module_local.fragment.AlbumsFragment
import com.hjl.module_local.fragment.AuthorFragment
import com.hjl.module_local.fragment.LocalFragment
import com.hjl.module_main.mvp.fragment.MusicService
import com.hjl.module_main.mvp.fragment.PlayFragment
import kotlinx.android.synthetic.main.activity_local_music.*

class LocalMusicActivity : BaseMultipleActivity() {

    val fragments : ArrayList<Fragment> = ArrayList()
    val mTitles : ArrayList<String> = ArrayList()
    var mPageAdapter : LazyFragmentPagerAdapter? = null

    var mBinder : MusicService.MusicBinder? = null

    private val permissions = arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.INTERNET, Manifest.permission.CAMERA)

    override fun initTitle() {
        mTitleCenterTv.text = resources.getString(R.string.local_app_name)
        mTitleLeftIv.visibility = View.INVISIBLE
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_local_music)

        initView()
        initData()

        PermissionsUtiles.requestPermissions(this, permissions) //请求权限
        val startMusicIntent = Intent(this, MusicService::class.java)
        bindService(startMusicIntent, connection,BIND_AUTO_CREATE)
        startService(startMusicIntent)
    }

    override fun initView() {


        mPageAdapter = LazyFragmentPagerAdapter(supportFragmentManager, fragments, mTitles)
        local_viewpager.adapter = mPageAdapter
        local_tab_layout.setupWithViewPager(local_viewpager)

    }

    override fun initData() {
        super.initData()

        mTitles.add("音乐")
        mTitles.add("专辑")
        mTitles.add("歌手")
    }

    private val connection = object : ServiceConnection {
        override fun onServiceConnected(name: ComponentName, service: IBinder) {
            mBinder = service as MusicService.MusicBinder

            fragments.add(LocalFragment.newInstance(mBinder))
            fragments.add(AlbumsFragment.newInstance(mBinder!!))
            fragments.add(AuthorFragment.newInstance(mBinder!!))
            mPageAdapter!!.notifyDataSetChanged()

            val playFragment = play_fragment as PlayFragment
            playFragment.setBinder(mBinder)


        }

        override fun onServiceDisconnected(name: ComponentName) {

        }
    }

    override fun onDestroy() {
        super.onDestroy()
        unbindService(connection)
    }
}
